cli/lib/kontena/cli/external_registries/add_command.rb

```diff
@@ -14,6 +14,8 @@ def execute
       require_current_grid
       token = require_token
 
+      self.url = "https://#{self.url}" unless self.url.start_with?('http')
```

Comment:
Could be: `self.url.prepend('https://') unless self.url.start_with?('http')`

Reply:
(and it could be in the param definition `do` block)

```ruby
parameter '[URL]', 'Docker Registry url', default: 'https://index.docker.io/v2/' do |url|
  url.prepend('https://') unless url.start_with?('http')
end
```

Reply:
I like `do` block but it needs to return always something.. and prepend modifies a string which is not very good with ruby 2.4 and later.

Reply:
Right, the block return is actually used, modifying the param does nothing.